controllers/registeredservice_controller.go

```diff
@@ -161,12 +167,184 @@ func (r *RegisteredServiceReconciler) addServiceToCatalog(ctx context.Context, s
+				AllowPrivilegeEscalation: &f,
+			},
+		}
+		cronjob.Spec = batchv1.CronJobSpec{
+			// run every n minutes
+			Schedule:                   fmt.Sprintf("*/%d * * * *", rs.Spec.HealthCheck.Container.Minutes),
+			ConcurrencyPolicy:          batchv1.ForbidConcurrent,
+			SuccessfulJobsHistoryLimit: &one,
+			FailedJobsHistoryLimit:     &one,
+			JobTemplate: batchv1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					ActiveDeadlineSeconds: &thirty,
+					BackoffLimit:          &two,
+					Parallelism:           &one,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers:    []corev1.Container{container},
+							RestartPolicy: corev1.RestartPolicyNever,
+							SecurityContext: &corev1.PodSecurityContext{
+								RunAsUser:    &uid,
+								RunAsGroup:   &uid,
+								RunAsNonRoot: &t,
+								SeccompProfile: &corev1.SeccompProfile{
+									Type: corev1.SeccompProfileTypeRuntimeDefault,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		return ctrl.SetControllerReference(rs, &cronjob, r.Scheme)
+	})
+
+	if err != nil {
+		return err
+	}
+
+	l := log.FromContext(ctx).WithValues("cronjob", cronjob.Name, "namespace", cronjob.Namespace)
+	if operation == controllerutil.OperationResultCreated {
+		l.Info("Created healthcheck cronjob")
+	} else {
+		l.Info("Modified healthcheck cronjob", "operation", operation)
+	}
+
+	return nil
+}
+
+func (r *RegisteredServiceReconciler) handleHealthcheck(ctx context.Context, rs *primazaiov1alpha1.RegisteredService) error {
+	l := log.FromContext(ctx)
+	cronjob := batchv1.CronJob{}
+	err := r.Get(ctx, types.NamespacedName{Name: rs.Name, Namespace: rs.Namespace}, &cronjob)
+	if k8errors.IsNotFound(err) {
+		l.Info("creating registered service healthcheck")
+		err := r.registerHealthcheck(ctx, rs)
+		rs.Status.State = primazaiov1alpha1.RegisteredServiceStateUnknown
+		return err
+	}
+
+	jobList := batchv1.JobList{}
+	err = r.List(ctx, &jobList, client.InNamespace(cronjob.Namespace))
+	if err != nil {
+		return err
+	}
+
+	completed := false
+	failed := false
+
+	for _, job := range jobList.Items {
+		found := false
+		for _, ownerReference := range job.OwnerReferences {
+			if ownerReference.UID == cronjob.UID {
+				found = true
+				break
+			}
+		}
+		if !found {
+			continue
+		}
+
+		for _, cond := range job.Status.Conditions {
+			if cond.Type == batchv1.JobComplete && cond.Status == corev1.ConditionTrue {
+				completed = true
+				break
+			} else if cond.Type == batchv1.JobFailed && cond.Status == corev1.ConditionTrue {
+				failed = true
+				break
+			}
+		}
+	}
+
+	l.Info("Job status", "completed", completed, "failed", failed)
+	if completed {
+		// only keep it in 'Claimed' if the healthcheck succeeded
+		if rs.Status.State == primazaiov1alpha1.RegisteredServiceStateClaimed {
+			rs.Status.State = primazaiov1alpha1.RegisteredServiceStateClaimed
+		} else {
+			rs.Status.State = primazaiov1alpha1.RegisteredServiceStateAvailable
+		}
```

Comment:
I don't get the complete logic you are checking if the state is claimed set it to claimed? Would it not be better to just check if state != claimed set it to available

Reply:
Fixed in d5f2d55.